core/src/test/java/org/apache/spark/memory/TaskMemoryManagerSuite.java

```diff
@@ -25,27 +26,37 @@
 
 public class TaskMemoryManagerSuite {
 
+  static TaskMemoryManager manager;
+
+  @After
+  public void after() {
+    Assert.assertEquals(0, manager.getMemoryConsumptionForThisTask());
```

Comment:
This sounds like this assert triggering, not the actual test failing?

I'm talking about is leaking memory *and* inserting a `fail("blah")` in a test and seeing what's reported.